src/nunit.analyzers/Helpers/AssertExpressionHelper.cs

```diff
@@ -52,32 +53,58 @@ internal static class AssertExpressionHelper
 
             foreach (var constraintPart in constraintParts)
             {
-                var invocations = constraintPart.SplitCallChain()
-                    .OfType<InvocationExpressionSyntax>()
-                    .Where(i => i.ArgumentList.Arguments.Count == 1);
+                var expected = GetExpectedArgumentFromConstraintPart(constraintPart, semanticModel, cancellationToken);
 
-                foreach (var invocation in invocations)
+                if (expected != null)
                 {
-                    var symbol = semanticModel.GetSymbolInfo(invocation, cancellationToken).Symbol;
-
-                    if (symbol is IMethodSymbol methodSymbol
-                        && methodSymbol.Parameters.Length == 1
-                        && methodSymbol.Parameters[0].Name == NunitFrameworkConstants.NameOfExpectedParameter)
-                    {
-                        var argument = invocation.ArgumentList.Arguments[0];
-                        expectedArguments.Add((argument.Expression, methodSymbol));
-                    }
+                    expectedArguments.Add(expected.Value);
```

Comment:
Previously we split expression only by `&`/`|` operators, now - by `.And`/`.Or` properties as well.
Because of that I assume there might be only one (or none) expected argument per part.
Should have same result, if I haven't missed anything

Reply:
Sounds reasonable